app/src/main/java/org/fossasia/openevent/adapters/TrackSessionsListAdpater.java

```diff
@@ -0,0 +1,50 @@
+package org.fossasia.openevent.adapters;
+
+import android.widget.Filter;
+
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.dbutils.DbSingleton;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import timber.log.Timber;
+
+/**
+ * Created by suheb on 17/2/16.
+ */
+public class TrackSessionsListAdpater extends SessionsListAdapter {
+
+    private int mTrackId;
+
+    @SuppressWarnings("all")
+    private Filter filter = new android.widget.Filter() {
+        @Override
+        protected FilterResults performFiltering(CharSequence constraint) {
+            DbSingleton instance = DbSingleton.getInstance();
+            String searchText = constraint.toString();
+            final ArrayList<Session> filteredSessionList = instance.getSessionbySearchTextAndTrack(searchText, mTrackId);
+            FilterResults filterResults = new FilterResults();
+            filterResults.values = filteredSessionList;
+            filterResults.count = filteredSessionList.size();
+            Timber.d("Filtering done total results %d", filterResults.count);
+            return filterResults;
+        }
+
+        @Override
+        protected void publishResults(CharSequence constraint, FilterResults results) {
+            animateTo((List<Session>) results.values);
+        }
+    };
+    public void setTrackId(int trackId) {
```

Comment:
I suppose you should perhaps override the constructor and accept the track value via that, although this is fine as well.